entrypoint.sh

```diff
@@ -4,21 +4,23 @@
 set -ex
 
 # Check whether there is a passwd entry for the container UID
-uid=$(id -u)
-gid=$(id -g)
-
-# turn off -e for getent because it will return error code in anonymous uid case
-set +e
-uidentry=$(getent passwd $uid)
-set -e
-
-# If there is no passwd entry for the container UID, attempt to create one
-if [[ -z "$uidentry" ]] ; then
-    if [[ -w /etc/passwd ]] ; then
-        echo "$uid:x:$uid:$gid:anonymous uid:$SPARK_HOME:/bin/false" >> /etc/passwd
-    else
-        echo "Container ENTRYPOINT failed to add passwd entry for anonymous UID"
-    fi
+myuid="$(id -u)"
+# If there is no passwd entry for the container UID, attempt to fake one
+# You can also refer to the https://github.com/docker-library/official-images/pull/13089#issuecomment-1534706523
+# It's to resolve OpenShift random UID case.
+# See also: https://github.com/docker-library/postgres/pull/448
+if ! getent passwd "$myuid" &> /dev/null; then
+    for wrapper in {/usr,}/lib{/*,}/libnss_wrapper.so; do
+      if [ -s "$wrapper" ]; then
+        NSS_WRAPPER_PASSWD="$(mktemp)"
+        NSS_WRAPPER_GROUP="$(mktemp)"
+        export LD_PRELOAD="$wrapper" NSS_WRAPPER_PASSWD NSS_WRAPPER_GROUP
+        mygid="$(id -g)"
+        printf 'spark:x:%s:%s:${SPARK_USER_NAME:-anonymous uid}:%s:/bin/false\n' "$myuid" "$mygid" "$SPARK_HOME" > "$NSS_WRAPPER_PASSWD"
```

Comment:
I think this line was supposed to look something like this:
```sh
printf 'spark:x:%s:%s:%s:%s:/bin/false\n' "$myuid" "$mygid" "${SPARK_USER_NAME:-anonymous uid}" "$SPARK_HOME" > "$NSS_WRAPPER_PASSWD"
```

Because currently the nss wrapper passwd looks something like:
```
spark:x:1000:1000:${SPARK_USER_NAME:-anonymous uid}:/foobar:/bin/false

```

the single quotes around the printf template prevent bash from interpolating $SPARK_USER_NAME